glitch/parsers/cmof.py

```
@@ -97,56 +97,92 @@ def __get_element_code(start_token, end_token, code):
+                var.variables = aux_vars
+            elif isinstance(v, ScalarNode):
+                create_variable(v, key.value, str(v.value), child)
+            elif isinstance(v, SequenceNode):
+                value = []
+                for i, val in enumerate(v.value):
+                    if isinstance(val, CollectionNode):
+                        var = create_variable(val,
+                                              f"{key.value}[{i}]", None, child)
+                        aux_vars = []
+                        aux_vars += AnsibleParser.__parse_vars(
+                            unit_block, f"{key.value}[{i}]", val, code)
+                        var.variables = aux_vars
+                    else:
+                        value.append(val.value)
+
+                if (len(value) > 0):
+                    create_variable(v, key.value, str(value), child)
+
+        variables = []
         if isinstance(token, MappingNode):
             for key, v in token.value:
                 if hasattr(key, "value") and isinstance(key.value, str):
-                    AnsibleParser.__parse_vars(unit_block, cur_name + key.value + ".", v, code)
+                    if cur_name == "":
+                        process_var(key, v)
+                    else:
+                        process_var(key, v, True)
                 elif isinstance(key.value, MappingNode):
                     AnsibleParser.__parse_vars(unit_block, cur_name, key.value[0][0], code)
-        elif isinstance(token, SequenceNode):
-            value = []
-            for i, v in enumerate(token.value):
-                if isinstance(v, CollectionNode):
-                    AnsibleParser.__parse_vars(unit_block, f"{cur_name[:-1]}[{i}].", v, code)
-                else:
-                    value.append(v.value)
+        elif isinstance(token, SequenceNode) or isinstance(token, ScalarNode):
+            pass  # can't happen, if it happens is not syntatically right so it ignores
```

Should it fail silently or should we raise an exception here?

---

Nuno advise me that this parsing is wrong, I am going to look at it again and fix it.